Events details (#5383)
rotki/rotki: docs/api.rst

   :resjson object result: A list of transaction entries to return for the given filter.
   :resjson object entry: A single transaction entry
   :resjson bool ignored_in_accounting: A boolean indicating whether this transaction should be ignored in accounting or not

>    :resjson list decoded_events: A list of decoded events for the given transaction. Each event is an object comprised of the event entry and a boolean denoting if the event has been customized by the user or not. Each entry also has a `has_details` flag. If `has_details` is true, then it is possible to call /history/events/details endpoint to retrieve some useful details about the event.

Reviewer: ```suggestion
   :resjson list decoded_events: A list of decoded events for the given transaction. Each event is an object comprised of the event entry and a boolean denoting if the event has been customized by the user or not. Each entry also has a `has_details` flag. If `has_details` is true, then it is possible to call /history/events/details endpoint to retrieve some extra information about the event.
```

   :statuscode 200: Custom asset types retrieved successfully.
   :statuscode 409: No user is currently logged in. Check error message.
   :statuscode 500: Internal rotki error.

Events Details
================

.. http:get:: /api/(version)/history/events/details

   Doing a GET on this endpoint will return the details of a history event.

   **Example Request**:

   .. http:example:: curl wget httpie python-requests

      POST /api/1/assets/custom HTTP/1.1
      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {"event_identifier": "0x66e3d2686193e01a4fbf0f598872236402edfe2f4efad84c4f6cdc753b8c78e3", "sequence_index": "222"}

>    :reqjson int[optional] event_identifier: The identifier of the event to be queried.

Reviewer: is not an integer, is a string. Also is not optional

   **Example Response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Content-Type: application/json

      {
          "result": {
              "sub-swaps": [

>                   {"from_amount": "100.0", "to_amount": "0.084", "from_asset": A_SUSHI.identifier, "to_asset": A_ETH.identifier},  # noqa: E501
>                   {"from_amount": "0.084", "to_amount": "98.2", "from_asset": A_ETH.identifier, "to_asset": A_USDT.identifier},  # noqa: E501

Reviewer 2: This is invalid json copy pasta Use the correct values and make sure that there is no trailing commas. Run make docs.

```suggestion
                  {"from_amount": "100.0", "to_amount": "0.084", "from_asset": A_SUSHI.identifier, "to_asset": A_ETH.identifier},  # noqa: E501
                  {"from_amount": "0.084", "to_amount": "98.2", "from_asset": A_ETH.identifier, "to_asset": A_USDT.identifier},  # noqa: E501
```

              ],
          },
          "message": ""
      }

>    :resjson object result: A dictionary with the details.

Reviewer 2: ```suggestion
   :resjson object result: A dictionary with the details. It may contain one of the following amount of details.
```

>               ],

Reviewer: the sequence index is optional. How is the response when there are multiples?

Author: No, it's not optional, it's required. It was a mistake in the docs

   :resjson object result: A dictionary with the details. If there is no details, an empty dictionary is returned.
   :resjson list sub-swaps: A list with the details of each sub-swap. Each entry contains the following keys: amount_in, amount_out, from_asset, to_asset.

   :statuscode 200: The details were returned successfully.
   :statuscode 400: Provided JSON is in some way malformed.

>    :statuscode 404: There is no event with the provided event_identifier and sequence_index.

Reviewer: ```suggestion
   :statuscode 404: There is no event with the provided event_identifier or sequence_index.
```

Author: Why this change? The events are matched by event_identifier + sequence_index

Reviewer: it was due to how the docs were. If it was a mistaken then it is an `and`. If it was optional then is `or`
